lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -1001,6 +1002,14 @@ SubExprInfo ExprEmitter::visitComb(ArraySliceOp op) {
   return {Selection, arrayPrec.signedness};
 }
 
+SubExprInfo ExprEmitter::visitComb(ArrayGetOp op) {
+  auto arrayPrec = emitSubExpr(op.input(), Selection);
+  os << '[';
+  emitSubExpr(op.index(), LowestPrecedence);
+  os << ']';
+  return {Selection, arrayPrec.signedness};
```

Comment:
How does signedness get propagated through verilog arrays?  Are you sure this is the right result signedness?

Reply:
As usual, the spec is unclear. Section 11.8.1 seems to govern it and does not explicitly mention multidimensional arrays. It only says:

> Bit-select results are unsigned, regardless of the operands. 

But this is not a simple bit-vector. And later on:

> — The sign and size of any self-determined operand are determined by the operand itself and
independent of the remainder of the expression. 
>  -- For non-self-determined operands, the following rules apply:
> • If any operand is real, the result is real.
> • If any operand is unsigned, the result is unsigned, regardless of the operator.
> • If all operands are signed, the result will be signed, regardless of operator, except when specified otherwise.

Neither Verilator nor Questa seem to care one way or another. I've tried all the permutations of signed/unsigned of the below code without a peep from either.

```
module test ();
logic signed [5:0][7:0] a;
logic unsigned [7:0] aprime = a[0];
endmodule
```

I think it's a reasonable assumption that this is correct, but I've no way of verifying that. If someone else knows otherwise or we run into problems, we can change it.